src/main/java/service/FruitShopService.java

```diff
@@ -0,0 +1,36 @@
+package service;
+
+import exception.OperationException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import model.FruitTransaction;
+import model.OperationType;
+import strategy.OperationHandler;
+
+public class FruitShopService {
+    private final Map<String, Integer> fruitInventory = new HashMap<>();
```

Comment:
Avoid using raw types in your code. The 'fruitInventory' should use a specific type for the key, such as 'Fruit' if you have a 'Fruit' class, to avoid potential class cast issues at runtime.